pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -95,7 +95,7 @@ let
 
       platforms =
         if args ? meta.platforms then
-          lib.intersectLists args.meta.platforms dotnet-sdk.meta.platforms
+          args.meta.platforms
```

Comment:
Actually the first issue with this code is that it should take the `meta.platforms` of the `dotnet-runtime` not `dotnet-sdk` right? (@corngood)

Also this is an artifact of when this PR was using self-contained, with which you can make a derivation for platforms that nixpkgs doesn't have a runtime package for (because it can download the runtime from nuget).

Reply:
I'm not sure we should even be intersecting the platforms here, or providing a default.  Are there examples of this in other languages?

It seems like we should just let the dependencies fail to eval.

Reply:
I don't know if there are examples of this in other languages, however I don't see the point in allowing platforms where the packages will not be able to run on.
If we can provide this safeguard without too much trouble I don't see why not.